### PR TITLE
Fz/date range input footer element

### LINKED ISSUE #5537: [DateRangeInput2]: footerElement prop does not render
#### Environment

- __Package version(s)__: "@blueprintjs/core": "^4.7.0",
    "@blueprintjs/datetime": "^4.3.4",
    "@blueprintjs/datetime2": "^0.5.0",
    "@blueprintjs/popover2": "^1.5.0",
    "@blueprintjs/select": "^4.5.2",
    "@blueprintjs/table": "^4.5.0",
    "@blueprintjs/timezone": "^4.4.4",
- __Operating System__: MacOs Monterey 12.1
- __Browser name and version__: Chrome Version 104.0.5112.101

#### Code Sandbox
[Example](https://codesandbox.io/s/blueprint-sandbox-forked-crmux7?file=/src/DatetimeExample.tsx)

#### Steps to reproduce

1. add footerElement prop to DateRangeInput2 
1. ensure that it does not render

#### Actual behavior

No footer component rendered

#### Expected behavior

Render footer component

#### Possible solution

It seems that the prop is not used at all by DateRangeInput or by DateRangeInput2. Both DateInput and DateRangeInput, do render it. I think it just needs to be added

### PR DESCRIPTION
**Pull request by fnziman**
#### Checklist

- [ ] Includes tests
- [ ] Update documentation

#### Changes proposed in this pull request:

Adds the footerElement to the DateRangePicker

#### Reviewers should focus on:

Seeing that the footer now works

#### Screenshot

image

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @fnziman! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: packages/datetime/src/dateRangePicker.tsx [hunk 1]
```diff
@@ -237,9 +237,12 @@ export class DateRangePicker extends AbstractPureComponent2<DateRangePickerProps
         return (
             <div className={classes}>
                 {this.maybeRenderShortcuts()}
-                <div>
-                    {this.renderCalendars(isShowingOneMonth)}
-                    {this.maybeRenderTimePickers()}
+                <div className={DateClasses.DATEPICKER_CONTENT}>
+                    <div>
```
Comment (adidahiya) [inline packages/datetime/src/dateRangePicker.tsx]: This doesn't seem like the ideal DOM structure... it probably makes more sense for the calendar(s) and time picker(s) to live adjacent to the footer element. I would like to see more options added to the docs example so we can test out this new layout. I'll try to push my ideas as a commit on your branch.

File: packages/datetime/src/dateRangePicker.tsx [hunk 2]
```diff
@@ -237,9 +237,12 @@ export class DateRangePicker extends AbstractPureComponent2<DateRangePickerProps
         return (
             <div className={classes}>
                 {this.maybeRenderShortcuts()}
-                <div>
-                    {this.renderCalendars(isShowingOneMonth)}
-                    {this.maybeRenderTimePickers()}
+                <div className={DateClasses.DATEPICKER_CONTENT}>
```
Comment (adidahiya) [inline packages/datetime/src/dateRangePicker.tsx]: also this CSS class was named for and designed for "datepicker", not "date range picker", so we'll need to make a new class to keep the naming semantics clear

File: packages/datetime/src/dateRangePicker.tsx (hunk 1 above)
Comment (fnziman) [reply packages/datetime/src/dateRangePicker.tsx]: I made an update that I think is more what you are looking for..

**Comment by adidahiya** [comment]
Rendering the footer element exposed a few layout issues with the multiple calendars and time pickers rendered by DateRangePicker. I also discovered an unrelated bug where it renders two time pickers even when `singleMonthOnly={true}`. The fix for all these things was not trivial, so I took your first commit and added my changes on it in a new PR: #5539

**Comment by adidahiya** [comment]
Thanks for the attempted PR -- I'm going to close this in favor of my new one, but you'll still get credited as a collaborator on the commit when that PR merges :)

**Pull request closed without merging**